### PR TITLE
Deprecate AIC, AICc, BIC in favor of aic, aicc, bic

### LINKED ISSUE #177: Naming of AIC, BIC, and AICc
The function names `AIC`, `BIC`, and `AICc` are presumably in all caps for similarity with how one would write them outside of code. However, it's a little confusing because folks familiar with Julia naming conventions might assume that these are types rather than functions, since types are typically in upper camel case and functions in lowercase.

Thus, I propose that `AIC`, `BIC`, and `AICc` be deprecated in favor of `aic`, `bic`, and `aicc`. If people think this is a good idea I'd be happy to submit a PR.

### PR DESCRIPTION
**Pull request by ararslan**
Fixes #177.

### CODE REVIEW

**Comment by ararslan** [comment]
The travis failures are unrelated.

File: src/statmodels.jl [hunk 1]
```diff
@@ -47,35 +47,35 @@ fit(obj::StatisticalModel, data...) = error("fit is not defined for $(typeof(obj
...
     n = nobs(obj)
     -2loglikelihood(obj) + 2k + 2k*(k+1)/(n-k-1)
 end
 
 """
-    BIC(obj::StatisticalModel)
+    bic(obj::StatisticalModel)
 
-Akaike's Information Criterion, defined as `-2 log L + k log n`, with `L`
+Bayesian Information Criterion, defined as `-2 log L + k log n`, with `L`
```
Comment (nalimilan) [inline src/statmodels.jl]: Good catch! Looks like I copied/pasted docstrings a bit too fast. :-)

**Comment by nalimilan** [comment]
Should we do the same for `R2` and `adjR2`?

**Comment by ararslan** [comment]
Yeah, that would make sense. If it were up to me I would have named it `rsquared` instead of `R2`, but that's beside the point. 😉

**Comment by nalimilan** [comment]
Why do you prefer `rsquared`? I find it less natural and it's much longer. `adjrsquared` is very hard to read. Actually, my preferred solution is `r²` and `adjr²`, but I have added the `2` variants for cases where `²` is hard to type.

**Comment by ararslan** [comment]
I think `rsquared` is more immediately descriptive than `r2` but I don't feel too strongly about it. I like `r²` but I'm honestly not sure how to type the superscript 2. Is there a LaTeX completion shortcut?

**Comment by nalimilan** [comment]
`²` is on my French keyboard, and I hadn't realized that it wasn't on standard QWERTY keyboards. Anyway, the LaTeX sequence is `\^2`.

**Comment by nalimilan** [comment]
Thanks!

**Comment by ararslan** [comment]
My pleasure! I love helping out with this stuff. :smile:

**Pull request merged**